### PR TITLE
util_linux: ensure that O_NOCTTY is raised when opening terminals

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

**Comment by tomponline** [comment]
Does this fix https://github.com/lxc/lxd/issues/9542?

**Comment by brauner** [comment]
> Does this fix #9542?

No, that's just a generic bugfix. :)

File: shared/util_linux.go [hunk 1]
```diff
@@ -330,14 +334,14 @@ func OpenPtyInDevpts(devpts_fd int, uid, gid int64) (*os.File, *os.File, error)
...
 			ptx = os.NewFile(uintptr(fd), "/dev/pts/ptmx")
 		}
 	} else {
-		ptx, err = os.OpenFile("/dev/ptmx", os.O_RDWR|unix.O_CLOEXEC, 0)
-		if err != nil {
-			return nil, nil, err
+		fd, err := unix.Openat(-1, "/dev/ptmx", os.O_RDWR|unix.O_CLOEXEC|NOCTTY, 0)
+		if err == nil {
+			ptx = os.NewFile(uintptr(fd), "/dev/pts/ptmx")
 		}
```
Comment (stgraber) [inline shared/util_linux.go:345]: Aren't we missing the error handling then? What happens if the Openat fails?

**Review by tomponline — APPROVED** [review]

File: shared/util_linux.go (hunk 1 above)
Comment (brauner) [reply shared/util_linux.go]: Fixed this up and rewrote parts.

**Pull request merged**